ABOS/ASFS/lftp_sync.sh

```diff
@@ -19,29 +18,46 @@ sync_files() {
     lftp -e "open -u $FTP_USER,$FTP_PASSWORD $FTP_SOURCE; mirror -e --parallel=10 $FTP_EXTRA_OPTS -vvv --log=$lftp_log_file $FTP_DIR $ABOS_SOFS_WIP_DIR_FTP/; quit"
 }
 
+# copy files to incoming dir with the correct permissions
+# $1 - subdirectory to copy to, relative to $INCOMING_DIR
+# "$@" - file(s) to copy
+copy_to_incoming() {
+    local incoming_dir="$INCOMING_DIR/$1" ; shift
+    local file_basename, tmp_file
+
+    for file; do
+        file_basename=$(basename "$file")
+        tmp_file=$(mktemp -t "${file_basename}.XXXXXX")
+        install -g projectofficer -m 0664 "$file" "$tmp_file" && \
+            mv "$tmp_file" "$incoming_dir/$file_basename"
+    done
+}
+
 # main - update SOFS files from BoM ftp site for given year
 # $1 - year (current year if not given)
 main() {
     local year=$1; shift
-    [ -z "$year" ] && year=`date +%Y`
+    local tmp_lftp_output_file, tmp_files_added
+
+    [ -z "$year" ] && year=$(date +%Y)
     declare -rg FTP_DIR="/register/bom404/outgoing/IMOS/MOORINGS/$year"
 
-    mkdir -p $ABOS_SOFS_WIP_DIR_FTP
-    local tmp_lftp_output_file=`mktemp`
-    sync_files $tmp_lftp_output_file || return 1
+    mkdir -p "$ABOS_SOFS_WIP_DIR_FTP"
+    tmp_lftp_output_file=$(mktemp)
```

Comment:
Can you add a trap here to ensure the file is deleted no matter what? 

Something like:
`trap rm -f $tmp_lftp_output_file EXIT`

Reply:
Done